### PR TITLE
Better offline messages in Comments list

### PR DESCRIPTION
**Pull request by ctarda**
In the spirit of small PRs, this one implements part of #5759 : comments list.

To test:
1. Enter flight mode
2. Navigate to any site > Comments
3. If there is content cached from a previous visit, it should be visible, but there should be an alert warning that connection is offline:
img_2605

4. If there is no content cached, there should be also an alert, and when the alert is dismissed, there should be an error message in the no results view, warning that the connection is offline.
5. In this case, pulling to refresh will display the same alert and the same error message.
img_2606
img_2607

6. After the connection is back up, pulling to refresh should load data again.

Open question. Should I try to make the content reload automatically after the connection is back up? I would like to give it a shot, but I am not sure if it is out of scope.

### CODE REVIEW

**Comment by ctarda** [comment]
I pushed an update:
- If there is locally cached content, display an alert.
- If there is no content, use `NoResultsView`

I think I finally got it! 😉

File: WordPress/Classes/Utility/ReachabilityUtils.h [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #import <Foundation/Foundation.h>
 
+NS_ASSUME_NONNULL_BEGIN
```
Comment (elibud) [inline WordPress/Classes/Utility/ReachabilityUtils.h]: TIL 👍

File: WordPress/Classes/ViewRelated/Comments/CommentsViewController.m [hunk 2]
```diff
@@ -407,22 +418,30 @@ - (void)syncHelper:(WPContentSyncHelper *)syncHelper syncContentWithUserInteract
...
+                                                failure(error);
+                                            });
+                                        }
+
+                                        if (![self isTableViewEmpty]) {
+                                            NSString *title = NSLocalizedString(@"Unable to Sync", @"Title of error prompt shown when a sync the user initiated fails.");
+                                            [WPError showNetworkingAlertWithError:error title:title];
+                                        }
+                                        if (userInteraction) {
+                                            double delayInSeconds = 0.2;
```
Comment (elibud) [inline WordPress/Classes/ViewRelated/Comments/CommentsViewController.m]: I'm a little lost here, why are we triggering a refresh?

**Comment by ctarda** [comment]
We are not triggering automatic reloads when we transition to a working connection yet. That is true for all the views I have been touching so far. I was planning on adding that (observing the status of the network connection, and reloading when it comes back up) in a separate PR, as we discussed on slack.

It seems like pulling to refresh after the connection is back up works. Would you mind trying again, please?

**Comment by elibud** [comment]
The problem is that the title for the `NoResultsView` is only set once, so, if the first time the view loads there wasn't an internet connection the title is set to: "The internet connection appears to be offline". Then if the connection is reestablished and we refresh the view, in a site with no comments, the `NoResultsView` is still present but the title is incorrect, it is never updated to `No comments yet`.

Adding this line:

`self.noResultsView.titleText = [self noResultsViewTitle];` 

in line `533` does the trick.

**Comment by elibud** [comment]
Marking this one as approved so we can start looking at https://github.com/wordpress-mobile/WordPress-iOS/pull/8836
Please make sure to add the missing line before merging.

**Comment by ctarda** [comment]
😭 I totally misunderstood your comment, sorry!

**Pull request merged**